src/crawlee/memory_storage_client/_memory_storage_client.py

```diff
@@ -45,13 +46,49 @@ class MemoryStorageClient(BaseStorageClient):
     _TEMPORARY_DIR_NAME = '__CRAWLEE_TEMPORARY'
     """Name of the directory used to temporarily store files during purges."""
 
-    def __init__(self, configuration: Configuration | None = None) -> None:
+    _DATASETS_DIR_NAME = 'datasets'
+    """Name of the directory containing datasets."""
+
+    _KEY_VALUE_STORES_DIR_NAME = 'key_value_stores'
+    """Name of the directory containing key-value stores."""
+
+    _REQUEST_QUEUES_DIR_NAME = 'request_queues'
+    """Name of the directory containing request queues."""
+
+    def __init__(
+        self,
+        configuration: Configuration | None = None,
+        *,
+        write_metadata: bool | None = None,
+        persist_storage: bool | None = None,
+        storage_dir: str | None = None,
+        default_request_queue_id: str | None = None,
+        default_key_value_store_id: str | None = None,
+        default_dataset_id: str | None = None,
+    ) -> None:
```

Comment:
I believe this isn't possible because the config relies on values from the env vars. Therefore, we should use the config values as defaults and override them with explicit arguments when provided.

Reply:
Maybe I wasn't clear enough. I mean something like this:

```py
def __init__(
    self,
    configuration: Configuration | None = None,
    *,
    write_metadata: bool,
    persist_storage: bool,
    storage_dir: str,
    default_request_queue_id: str,
    default_key_value_store_id: str,
    default_dataset_id: str,
) -> None:
    pass

def from_configuration(cls, configuration: Configuration | None = None) -> Self:
    return cls(
        write_metadata=configuration.write_metadata,
        persist_storage=configuration.persist_storage,
        storage_dir=configuration.storage_dir,
        default_request_queue_id=configuration.default_request_queue_id,
        default_key_value_store_id=configuration.default_key_value_store_id,
        default_dataset_id=configuration.default_dataset_id,
    )
```

...and if someone wants to override `Configuration` values with whatever, they can use `Configuration.model_copy(update={...})`. Now as I'm writing it, I don't really like it that much. Maybe we could add something like `Configuration.replace()` as a shortcut.

Reply:
Done